### PR TITLE
ARM OSX Migrator - Patch 1

### PR DESCRIPTION
**Pull request by shadowwalkersb**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by shadowwalkersb** [comment]
cc @isuruf

```
2021-01-14T20:46:00.5434390Z Undefined symbols for architecture arm64:
2021-01-14T20:46:00.5473120Z   "_lutil_memcmp", referenced from:
2021-01-14T20:46:00.5497980Z       _ldap_result in result.o
2021-01-14T20:46:00.5525070Z       _ldap_host_connected_to in os-ip.o
2021-01-14T20:46:00.5552070Z       _ldap_dn2domain in dnssrv.o
2021-01-14T20:46:00.5579360Z       _tlso_session_chkhost in tls_o.o
2021-01-14T20:46:00.5607230Z       _ldif_sput_wrap in ldif.o
2021-01-14T20:46:00.5633800Z ld: symbol(s) not found for architecture arm64
```

File: recipe/build.sh [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
 
 # disable server components
 ./configure \
     --prefix=$PREFIX  \
     --disable-slapd \
-    --disable-slurpd || { cat config.log; exit 1; }
+    --disable-slurpd \
+    --with-yielding_select=yes || { cat config.log; exit 1; }
+
+sed -i.bak 's|#define NEED_MEMCMP_REPLACEMENT 1|/* #undef NEED_MEMCMP_REPLACEMENT */|' include/portable.h
```
Comment (shadowwalkersb) [inline recipe/build.sh:14]: I am not sure this is the right solution, but can't find anything else that's working on google.

**Comment by shadowwalkersb** [comment]
@conda-forge/openldap, ready for review.

File: recipe/build.sh (hunk 1 above)
Comment (isuruf) [reply recipe/build.sh:14]: You can do `export ac_cv_func_memcmp_working=yes`

Comment (shadowwalkersb) [reply recipe/build.sh]: Done.

Comment (shadowwalkersb) [reply recipe/build.sh]: You think this is ready to go?

**Comment by shadowwalkersb** [comment]
ping @conda-forge/openldap

**Comment by bollwyvl** [comment]
Looks good: I have no way to evaluate how it will work out, but trust @shadowwalkersb will report back if something's extremely broken!

**Comment by shadowwalkersb** [comment]
Thanks for merging!

**Pull request merged**